frontend/src/store/cloudProfile/index.js

```diff
@@ -405,14 +411,13 @@ export const useCloudProfileStore = defineStore('cloudProfile', () => {
           continue
         }
 
-        logger.error(`Skipped machine image ${machineImage.name} as version ${versionObj.version} is not a valid semver version and cannot be normalized`)
+        logger.error(`Skipped machine image ${name} as version ${versionObj.version} is not a valid semver version and cannot be normalized`)
```

Comment:
I think the main Problem here is that the machine image getter is not cached (because of param). That's why we get the log message multiple times. We should solve the root problem. @holgerkoser removed it in his refactoring PR afaik

Reply:
I have moved the flattening of the machineImages into an own function and do after fetching and not over and over in the getter. The getter now directly returns the already flattened machine images.

```js
  async function fetchCloudProfiles () {
    const response = await api.getCloudProfiles()
    for (const cloudProfile of response.data) {
      set(cloudProfile, 'data.machineImages', flattenMachineImages(get(cloudProfile, 'data.machineImages')))
    }
    list.value = response.data
  }
  
  function flattenMachineImages (machineImages) {
    return flatMap(machineImages, machineImage => {
      const versions = filter(machineImage.versions, ({ version, expirationDate }) => {
        if (!semver.valid(version)) {
          logger.warn(`Skipped machine image ${machineImage.name} as version ${version} is not a valid semver version`)
          return false
        }
        return true
      })
      versions.sort((a, b) => {
        return semver.rcompare(a.version, b.version)
      })

      const name = machineImage.name
      const vendorName = vendorNameFromImageName(machineImage.name)
      const vendorHint = findVendorHint(configStore.vendorHints, vendorName)

      return map(versions, ({ version, expirationDate, cri, classification, architectures }) => {
        if (isEmpty(architectures)) {
          architectures = ['amd64'] // default if not maintained
        }
        return decorateClassificationObject({
          key: name + '/' + version,
          name,
          version,
          cri,
          classification,
          expirationDate,
          vendorName,
          icon: vendorName,
          vendorHint,
          architectures,
        })
      })
    })
  }
  
  function machineImagesByCloudProfileName (cloudProfileName) {
    const cloudProfile = cloudProfileByName(cloudProfileName)
    return get(cloudProfile, 'data.machineImages')
  }
  ```